igvc_ws/src/igvc_nav/src/igvc_nav_node.py

```diff
@@ -1,48 +1,81 @@
-pp.add_point(-97.4386215209961, 35.1950436830485)
-pp.add_point(-97.4388850511561, 35.1950436830485)
-pp.add_point(-97.4388850511561, 35.194881439209)
-pp.add_point(-97.4387313252294, 35.194935520488)
-pp.add_point(-97.4386215209961, 35.194881439209)
-pp.add_point(-97.4386215209961, 35.1949625611)
 
 def ekf_update(data):
     global pos, heading
-    pos = (data.longitude * 180/math.pi, data.latitude * 180/math.pi)
-    heading = data.global_heading * 180/math.pi
 
-def path_update(data):
-    global pp
-    pp.set_points([(_point.x, _point.y) for _point in data.points])
+    # position = data.pose.pose.position
+    # orientation = data.pose.pose.orientation
+
+    # pos = (position.x, position.y)
+    # quaternion = (
+    #     orientation.x,
+    #     orientation.y,
+    #     orientation.z,
+    #     orientation.w)
+    # heading = math.degrees(tf.transformations.euler_from_quaternion(quaternion)[2])
+
+    pos = (data.x, data.y)
+    heading = 360 - math.degrees(data.global_heading)
+
+def global_path_update(data):
+    points = [x.pose.position for x in data.poses] # Get points from Path
+    pp.set_points([(_point.x, _point.y) for _point in points]) # Give PurePursuit the points
 
 def timer_callback(event):
     if pos is None or heading is None:
         return
+
+    cur_pos = (pos[0], pos[1])
+
     lookahead = None
-    radius = 0.00002
-    while lookahead is None:
-        lookahead = pp.get_lookahead_point(pos[0], pos[1], radius) # radius about 1 meter
-        radius *= 1.5
+    radius = 0.3 # Start with a radius of 0.1 meters
+
+    while lookahead is None and radius <= 4: # Look until we hit 2 meters max
+        lookahead = pp.get_lookahead_point(cur_pos[0], cur_pos[1], radius)
+        radius *= 1.2
+    
+    if SHOW_PLOTS:
+        plt.figure(2)
+        plt.clf()
+        plt.xlim([-10, 10])
+        plt.ylim([-10, 10])
+        plt.plot(cur_pos[0], cur_pos[1], 's', markersize=16)
+
+        for point in pp.path:
+            plt.plot(point[0], point[1], '.', markersize=8)
+
     if lookahead is not None:
-        heading_to_la = 90 - math.atan2(lookahead[1] - pos[1], lookahead[0] - pos[0]) * 180 / (math.pi)
+
+        if SHOW_PLOTS:
+            plt.plot(lookahead[0], lookahead[1], 'x', markersize=16)
+
+        heading_to_la = math.degrees(math.atan2(lookahead[1] - cur_pos[1], lookahead[0] - cur_pos[0]))
         if heading_to_la < 0:
             heading_to_la += 360
 
-        # print('my x: ' + str(pos[0]))
-        # print('my y: ' + str(pos[1]))
+        # print('my x: ' + str(round(cur_pos[0], 2)))
+        # print('my y: ' + str(round(cur_pos[1], 2)))
 
-        # print('loc x: ' + str(lookahead[0]))
-        # print('loc y: ' + str(lookahead[1]))
+        # print('loc x: ' + str(round(lookahead[0], 2)))
+        # print('loc y: ' + str(round(lookahead[1], 2)))
 
         # print('want to move x:' + str(lookahead[0] - pos[0]))
         # print('want to move y:' + str(lookahead[1] - pos[1]))
```

Comment:
if this stuff is important, it would be nice to have a SHOW_PRINTS if statement instead of commenting these out.

If this isn't important, we should remove it.